crates/interpreter/src/valid.rs

```diff
@@ -195,7 +155,7 @@ impl<'m> Context<'m> {
             check(parser.is_empty())?;
         }
         self.check_section(parser, SectionId::Custom)?;
-        check(parser.is_empty())
+        check_and_return(parser.is_empty(), vec![])
```

Comment:
You don't need this `check_and_return` function. The following is much more readable by having small things do simple but precise things, and relying on composition to build more complex things (like unix philosophy).

```suggestion
        check(parser.is_empty())?;
        Ok(vec![]) // TODO(dev/fast-interp): implement
```